python_modules/dagster/dagster/core/asset_defs/decorators.py

```diff
@@ -106,12 +106,16 @@ def asset(
 
     Args:
         name (Optional[str]): The name of the asset.  If not provided, defaults to the name of the
-            decorated function.
+            decorated function. The asset's name must be a valid name in dagster (ie only contains
+            letters, numbers, and _) and may not contain python reserved keywords.
         namespace (Optional[Sequence[str]]): **Deprecated (use `key_prefix`)**. The namespace that
-            the asset resides in.  The namespace + the name forms the asset key.
+            the asset resides in.  The namespace + the name forms the asset key. The asset key must
```

Comment:
This can be a sequence, right? I think this description is a little bit confusing with respect to that.